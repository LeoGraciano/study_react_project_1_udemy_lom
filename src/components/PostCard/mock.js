export const postCardPropsMock = {
    title: 'Test Title 1',
    body: 'Test Body 1',
    id: '1',
    cover: 'img/img.png'
}