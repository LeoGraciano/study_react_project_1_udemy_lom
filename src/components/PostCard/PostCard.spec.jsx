import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { PostCard } from '.';
import { postCardPropsMock } from './mock';

const props = postCardPropsMock

describe('<PostCard />', () => {

  it('should render PostCard correctly', () => {
    expect.assertions(4)
    // const { debug } = render(<PostCard {...props} />);
    // debug();
    render(<PostCard {...props} />);
    expect(screen.getByRole('img')).toHaveAttribute('src', props.cover);
    expect(screen.getByAltText(props.title)).toHaveAttribute('alt', props.title);
    expect(screen.getByRole('heading', props.title)).toBeInTheDocument();
    expect(screen.getByText(props.body)).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot()
  });
});
