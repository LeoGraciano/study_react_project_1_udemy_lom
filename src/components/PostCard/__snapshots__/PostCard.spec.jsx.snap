// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Test Title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Test Title 1
    </h2>
    <p>
      Test Body 1
    </p>
  </div>
</div>
`;
